src/Control/Eff/QuickStart.hs

```diff
@@ -0,0 +1,226 @@
+import           Control.Eff.Exception
+import           Control.Monad                            ( when )
+
+
+-- | an effectful function that can throw an error
+--
+-- @
+-- tooBig = do
+--   when (i > 100) $ throwError $ show i
+--   return i
+-- @
+tooBig :: Member (Exc String) r => Int -> Eff r Int
+tooBig i = do
+  when (i > 100) $ throwError $ show i
+  return i
+
+-- | run the @tooBig@ effect based on a provided Int.
+--
+-- @
+-- runTooBig i = run . runError $ tooBig i
+-- @
+runTooBig :: Int -> Either String Int
+runTooBig i = run . runError $ tooBig i
+
+-- | run the @tooBig@ effect, giving @1@ as input
+--
+-- @
+-- runTooBig1 = runTooBig 1
+-- -- Right 1
+-- @
+runTooBig1 :: Either String Int
+runTooBig1 = runTooBig 1 -- Right 1
+
+-- | run the @tooBig@ effect, giving @200@ as input, which is too big for this
+-- effect.
+--
+-- @
+-- runTooBig1 = runTooBig 200
+-- -- Left "200"
+-- @
+runTooBig200 :: Either String Int
+runTooBig200 = runTooBig 200 -- Left "200"
+
+-- | an effectul computation using state. The state is of type @[Int]@.
+-- This function takes the head off the list, if it is there and return it.
+-- If state is the empty list, then it stays the same and returns @Nothing@.
+--
+-- @
+-- popState = do
+--  stack <- get
+--  case stack of
+--    []       -> return Nothing
+--    (x : xs) -> do
+--      put xs
+--      return $ Just x
+-- @
+popState :: Member (State [Int]) r => Eff r (Maybe Int)
+popState = do
+  stack <- get
+  case stack of
+    []       -> return Nothing
+    (x : xs) -> do
+      put xs
+      return $ Just x
+
+-- | run the popState effectful computation based on initial state. The
+-- result-type is the result of the computation @Maybe Int@ together with the
+-- state at the end of the computation @[Int]@
+--
+-- @
+-- runPopState xs = run . runState xs $ popState
+-- @
+runPopState :: [Int] -> (Maybe Int, [Int])
+runPopState xs = run . runState xs $ popState
+
+-- | run the popState effectful function
+--
+-- @
+-- runPopState123 = runPopState  [1, 2, 3]
+-- -- (Just 1, [2, 3])
+-- @
+runPopState123 :: (Maybe Int, [Int])
+runPopState123 = runPopState [1, 2, 3] -- (Just 1, [2, 3])
+
+-- | run the popState effectful function
+--
+-- @
+-- runPopStateEmpty = runPopState []
+-- -- (Nothing, [])
+-- @
+runPopStateEmpty :: (Maybe Int, [Int])
+runPopStateEmpty = runPopState [] -- (Nothing, [])
+
+-- | an effect that returns a number one more than the given
+--
+-- @
+-- oneMore = do
+--   x <- ask -- query the environment
+--   return $ x + 1 -- add one to the asked value and return it
+-- @
+oneMore :: Member (Reader Int) r => Eff r Int
+oneMore = do
+  x <- ask
+  return $ x + 1
+
+-- | Run the @oneMore@ effectful function by giving it a value to read.
+--
+-- @
+-- runOneMore = run . runReader 1 $ oneMore
+-- -- 2
+-- @
+-- TODO: runOneMore function and separate runOneMore1
+runOneMore1 :: Int
+runOneMore1 = run . runReader 1 $ oneMore -- 2
+
+-- | An effectful computation with multiple effects:
```

Comment:
@power-fungus would it be better if this computation was reusing the effect fragments before? or was this specifically avoided for some reason?